packages/xo-web/src/xo-app/sr/tab-general.js

```diff
@@ -62,8 +62,9 @@ const UsageTooltip = decorate([
       vdisUsage: (_, { group: { vdis } }) => formatSize(sumBy(vdis, 'usage')),
       snapshotsUsage: (_, { group: { snapshots } }) =>
         formatSize(sumBy(snapshots, 'usage')),
-      vmNamesByVdi: createCollectionWrapper(({ vdis }, { vbds, vms }) =>
-        mapValues(vdis, vdi => get(() => vms[vbds[vdi.VBD].VM]))
+      vbdsByVdi: (_, { vbds }) => keyBy(vbds, 'VDI'),
+      vmNamesByVdi: createCollectionWrapper(({ vbdsByVdi, vdis }, { vms }) =>
+        mapValues(vdis, vdi => get(() => vms[vbdsByVdi[vdi.id].VM].name_label))
```

Comment:
`vms[vbds[vdi.$VBDs[0]].VM]).name_label` ?

Reply:
We need to display VM name and already we have `vmNamesByVdi`.

Reply:
Now, I take `$ VBDs[0]` if the attached `VBD` doesn't exist.

Reply:
I need more information about the need.
Did we want to handle the case of a VDI attached to multiple VMs? 
- *No*: this test is not needed selecting the first VBD is enough.
- *Yes*: 
  - Should we show all VMs ?
  - Should we show only connected VMs?

Reply:
It's ok. I take the `$VBDs[0]`.